src/Microsoft.Extensions.Logging.AzureWebApps/Internal/BackgroundSink.cs

```
@@ -0,0 +1,113 @@
+        public BackgroundSink(ILogEventSink innerSink, int maxQueueSize)
+        {
+            if (innerSink == null)
+            {
+                throw new ArgumentNullException(nameof(innerSink));
+            }
+
+            _disposedToken = _disposedTokenSource.Token;
+
+            if (maxQueueSize <= 0)
+            {
+                _messages = new BlockingCollection<LogEvent>(new ConcurrentQueue<LogEvent>());
+            }
+            else
+            {
+                _messages = new BlockingCollection<LogEvent>(new ConcurrentQueue<LogEvent>(), maxQueueSize);
+            }
+
+            _innerSink = innerSink;
+
+            _workerThread = new Thread(Worker);
+            _workerThread.Name = GetType().Name;
+            _workerThread.IsBackground = true;
+            _workerThread.Start();
+        }
+
+        /// <inheritdoc />
+        public void Emit(LogEvent logEvent)
+        {
+            if (!_disposedToken.IsCancellationRequested)
+            {
+                _messages.Add(logEvent);
+            }
+        }
+
+        /// <summary>
+        /// Disposes this object instance.
+        /// </summary>
+        public virtual void Dispose()
+        {
+            if (!_disposedTokenSource.IsCancellationRequested)
+            {
+                lock (_disposedTokenSource)
+                {
+                    if (!_disposedTokenSource.IsCancellationRequested)
+                    {
+                        _disposedTokenSource.Cancel();
+                    }
+
+                    // Wait for the thread to complete before disposing the resources
+                    _workerThread.Join();
+                    _messages.Dispose();
+                }
+            }
+        }
+
+        private void Worker()
+        {
+            try
+            {
+                foreach (var logEvent in _messages.GetConsumingEnumerable(_disposedToken))
+                {
+                    PassLogEventToInnerSink(logEvent);
```

Ignore any exceptions thrown from here too? so that we can proceed with further log events...